### PR TITLE
fix(cli): restore permission check on workers

### LINKED ISSUE #8120: No permission checks for worker startup
```ts
// foo.ts
new Worker(new URL("./worker.ts", import.meta.url).toString(), { type: "module" });
```

```ts
// worker.ts
console.log("foo");
```

### Actual

```
$ deno run foo.ts
foo
```

### Expected

```
$ deno run foo.ts
PermissionDenied: ...
```

This is only happens on master. This is a 1.5.0 blocker.

### PR DESCRIPTION
**Pull request by kitsonk**
Fixes #8120

### CODE REVIEW

File: cli/tests/error_worker_permissions_local.ts.out [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[WILDCARD]
+error: Uncaught (in worker "") read access to "[WILDCARD]worker_types.ts", run again with the --allow-read flag
```
Comment (lucacasonato) [inline cli/tests/error_worker_permissions_local.ts.out:2]: This error message really isn't great. It don't actually specifically state that the permission was denied. Something we should address in a follow up PR.

File: cli/tests/error_worker_permissions_remote.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const w = new Worker(
+  new URL("http://localhost:4545/cli/tests/subdir/worker_types.ts").toString(),
```
Comment (lucacasonato) [inline cli/tests/error_worker_permissions_remote.ts:2]: The `new URL().toString()` isn't really needed here.

**Review by lucacasonato — APPROVED** [review]
LGTM. Nice to have tests for this now.

File: cli/tests/error_worker_permissions_local.ts.out (hunk 1 above)
Comment (nayeemrmn) [reply cli/tests/error_worker_permissions_local.ts.out:2]: It should say `error: Uncaught (in worker "") PermissionDenied: read access to ...`. Not sure if this PR did something to omit it or it's an earlier regression.

File: cli/tests/error_worker_permissions_remote.ts (hunk 2 above)
Comment (kitsonk) [reply cli/tests/error_worker_permissions_remote.ts:2]: 🤦

File: cli/tests/error_worker_permissions_local.ts.out (hunk 1 above)
Comment (kitsonk) [reply cli/tests/error_worker_permissions_local.ts.out:2]: It isn't this PR... I am not touching the error:

```rust
    if target_lib == TargetLib::Worker {
      runtime_permissions.check_specifier(&specifier)?;
    }
```

So very appropriate to fix in a follow up PR.

File: test_util/src/lib.rs [hunk 3]
```diff
@@ -880,7 +880,7 @@ impl CheckOutputIntegrationTest {
     reader.read_to_string(&mut actual).unwrap();
 
     let status = process.wait().expect("failed to finish process");
-    let exit_code = status.code().unwrap();
+    let exit_code = status.code().unwrap_or(0);
```
Comment (lucacasonato) [inline test_util/src/lib.rs:883]: I don't like this. It can potentially swallow errors in tests.

Comment (kitsonk) [reply test_util/src/lib.rs:883]: Then why did this occur, only on Ubuntu? https://github.com/denoland/deno/runs/1306229988#step:14:3856

Comment (lucacasonato) [reply test_util/src/lib.rs:883]: I don't know. The status code is a `None` value when the process was terminated by a signal. See https://doc.rust-lang.org/std/process/struct.ExitStatus.html

It is not safe to assume that it exited cleanly (code 0) on a `None` code.

Comment (kitsonk) [reply test_util/src/lib.rs]: Ok, I have removed it... see what happens this time with CI.

Comment (kitsonk) [reply test_util/src/lib.rs]: Same problem: https://github.com/denoland/deno/pull/8123/checks?check_run_id=1306322095#step:14:3858

It isn't occurring locally on my mac, and I don't know how the test case could be invalid.

**Comment by lucacasonato** [comment]
I'll test on Linux tomorrow morning. It looks like we aren't exiting cleanly.

File: cli/program_state.rs [hunk 4]
```diff
@@ -121,13 +121,19 @@ impl ProgramState {
     self: &Arc<Self>,
     specifier: ModuleSpecifier,
     target_lib: TargetLib,
-    dynamic_permissions: Permissions,
+    runtime_permissions: Permissions,
```
Comment (bartlomieju) [inline cli/program_state.rs:124]: I like this name much better

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/tests/error_worker_permissions_local.ts.out (hunk 1 above)
Comment (nayeemrmn) [reply cli/tests/error_worker_permissions_local.ts.out:2]: I've kind of fixed this locally, but I think it's a wontfix. There are error `Results` throughout cli which are passed to JS in some cases and terminated upon in Rust in other cases (like this one), and there isn't really a good way of selectively unifying the output that always works the way we want it to. The issue has been present for a while, it's mostly fine as it is.

**Pull request merged**